### ISSUE TITLE
CA1416 False positive when returning after failed inverted platform guard

### ISSUE THREAD

**Issue by RobSiklos** [dotnet/roslyn-analyzers#5814]
### Analyzer

**Diagnostic ID**: [CA1416](https://docs.microsoft.com/en-ca/dotnet/fundamentals/code-analysis/quality-rules/ca1416): `Validate platform compatibility`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 6.0.0 (Latest)

### Describe the bug

The analyzer says that the call site is reachable on all platforms. However, this is not correct. The platform has been checked earlier in the method and an exception is thrown if it's not the correct platform.

### Steps To Reproduce

Run the analyzer on the following code:

```csharp
public void M()
{
    if (OperatingSystem.IsWindows() == false)
    {
        throw new PlatformNotSupportedException();
    }

    // CA1416 warning for the following line.
    this.DoWin32Stuff();
}

[SupportedOSPlatform("windows")]
private void DoWin32Stuff() { }
```

### Expected behavior
No warning violation.

### Actual behavior
CA1416 warning violation.

**Comment by buyaa-n** [comment]
Duplicate of https://github.com/dotnet/roslyn-analyzers/issues/5771
